File: solutions/cpp/raindrops/1/raindrops.cpp
#include <sstream>
#include "raindrops.h"
namespace raindrops {
  auto convert(int n) -> string {
    auto ss = stringstream{};
    if(n % 3 == 0) ss << "Pling";
    if(n % 5 == 0) ss << "Plang";
    if(n % 7 == 0) ss << "Plong";
    return ss.str().size() ? ss.str() : to_string(n);
  }
}

File: solutions/vbnet/raindrops/1/Raindrops.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq

Public Module Raindrops
    Private sounds As IDictionary(Of Integer, String) = New Dictionary(Of Integer, String) From {
        {3, "Pling"},
        {5, "Plang"},
        {7, "Plong"}
    }

    Public Function Convert(number As Integer) As String
        Return String.Join("", sounds.Where(Function(x) number Mod x.Key = 0) _
                                     .Select(Function(x) x.Value) _
                                     .DefaultIfEmpty(number.ToString()))
    End Function
End Module